server/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java

```diff
@@ -90,6 +97,26 @@ void validateOpType(String opType) {
         }
     }
 
+    final class AutoIdHandler extends BaseRestHandler {
+        protected AutoIdHandler() {
+        }
+
+        @Override
+        public String getName() {
+            return "document_create_action";
+        }
+
+        @Override
+        public RestChannelConsumer prepareRequest(RestRequest request, final NodeClient client) throws IOException {
+            assert request.params().get("id") == null : "non-null id: " + request.params().get("id");
+            if (request.params().get("op_type") == null && clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.CURRENT)) {
```

Comment:
The rest-api-spec (`index.json`) says default for `op_type` is `index`, I think we should change that to be `create`.

I wonder if we should have an end-goal of removing `op_type` support here as a breaking change in 8.0?

Reply:
The problem is that this API spec mixes auto-id which non-auto-id. In one case, the default op_type is create, in the other case it is index. I have moved the explanation of the default into the description (as we have for many other parameters).